File: src/0134-Gas-Station/0134.cpp
#include <vector>

static int x = []() {std::ios::sync_with_stdio(false); cin.tie(0); return 0; }();
class Solution 
{
public:
    int canCompleteCircuit(vector<int>& gas, vector<int>& cost) 
    {
        int res = -1, all_sum = 0, min_sum = INT_MAX;
        for (int i = 0; i < gas.size(); ++i)
        {
            all_sum += gas[i] - cost[i];
            if (min_sum > all_sum) 
            {
                min_sum = all_sum;
                res = (i + 1)%gas.size();
            }
        }
        if (all_sum < 0) return -1;
        return res;
    }
};

File: src/0134-Gas-Station/0134.py
class Solution:
    def canCompleteCircuit(self, gas, cost):
        """
        :type gas: List[int]
        :type cost: List[int]
        :rtype: int
        """    
        res, all_sum, min_sum = -1, 0, float('inf')
        n = len(gas)
        for i in range(n):
            all_sum += gas[i] - cost[i]
            if min_sum > all_sum:
                min_sum = all_sum
                res = (i + 1)%n
                
        if all_sum < 0:
            return -1
        
        return res